src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortedGoods {
+  Alpha,
+  None,
+  Length,
+}
+
+type Options = {
+  sortedGoods: SortedGoods,
+  isGoodReversed: boolean,
+};
+
+function getGoodsSorted(
+  goods: string[],
+  { sortedGoods, isGoodReversed }: Options,
+) {
+  const copyGoods = [...goods];
+
+  if (sortedGoods === SortedGoods.Alpha) {
+    copyGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortedGoods === SortedGoods.Length) {
+    copyGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isGoodReversed) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState(SortedGoods.None);
+  const [isGoodReversed, setIsGoodReversed] = useState(false);
+  const sorted = getGoodsSorted(
+    goodsFromServer,
+    { sortedGoods, isGoodReversed },
+  );
+
+  const hadnlingReset = () => {
+    setSortedGoods(SortedGoods.None);
+    setIsGoodReversed(false);
+  };
+
+  const isResetButtonVisible = sortedGoods !== SortedGoods.None
+    || isGoodReversed;
+
+  const makeGoodsSortedAlpha = () => setSortedGoods(SortedGoods.Alpha);
```

Comment:
```suggestion
  const handleSort = (sortType: SortedGoods) => setSortedGoods(sortType);
```
better to use one handler with argument